model/pod/podutil.go

```diff
@@ -0,0 +1,105 @@
+package pod
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/pkg/errors"
+)
+
+// Constants representing default curl retry arguments.
+const (
+	curlDefaultNumRetries = 10
+	curlDefaultMaxSecs    = 100
+)
+
+// CurlCommand returns the command to curl the evergreen client and retries the request with the default retry parameters.
+func (p *Pod) CurlCommand(settings *evergreen.Settings) ([]string, error) {
+	var retryArgs string
+	if curlDefaultNumRetries != 0 && curlDefaultMaxSecs != 0 {
+		retryArgs = " " + curlRetryArgs(curlDefaultNumRetries, curlDefaultMaxSecs)
+	}
+	cmds, err := p.curlCommands(settings, retryArgs)
+	if err != nil {
+		return []string{}, errors.WithStack(err)
+	}
+	return []string{"CMD-SHELL", strings.Join(cmds, " && ")}, nil
+}
+
+// AgentCommand returns the arguments to start the agent.
+func (p *Pod) AgentCommand(settings *evergreen.Settings) []string {
+	return []string{
+		fmt.Sprintf("./%s", p.BinaryName()),
+		"agent",
+		fmt.Sprintf("--api_server=%s", settings.ApiUrl),
+		fmt.Sprintf("--mode=pod"),
+		fmt.Sprintf("--pod_id=%s", p.ID),
+		fmt.Sprintf("--pod_secret=%s", p.Secret),
+		fmt.Sprintf("--log_prefix=%s", filepath.Join(p.TaskContainerCreationOpts.WorkingDir, "agent")),
+		fmt.Sprintf("--working_directory=%s", p.TaskContainerCreationOpts.WorkingDir),
+		"--cleanup",
```

Comment:
I think we actually don't need this `--cleanup` flag. This flag tells the agent to clean up some leftover state in between executions of different tasks (e..g left over files in the file system). However, since pods don't have to be cleaned up between different tasks (since they're dedicated to the one task that runs it), we don't need this flag.